### ISSUE TITLE
Stacks SDK: Add clarity contract smart contract helper functions

### ISSUE THREAD

**Issue by jferrant** [stacks-network/sbtc#55]
Add ability to call read only functions and ability to construct stacks transactions for mutable functions with a vector of args

Definition of Done:
There exists a function to call a read only smart contract function 
There exists the necessary helper functions/function to call a modifying smart contract function

**Comment by friedger** [comment]
Something like this? https://github.com/stacks-network/stacks-blockchain/pull/3989

Where is the sdk hosted eventually? Here or on stacks-blockchain?

**Comment by jferrant** [comment]
> Something like this? [stacks-network/stacks-blockchain#3989](https://github.com/stacks-network/stacks-blockchain/pull/3989)
> 
> Where is the sdk hosted eventually? Here or on stacks-blockchain?


Yeah, pretty much exactly like that. I think we don't want stacks-blockchain to host this code because of its GPL license.  However, we can only use the stacks-sdk if we probably remove a bunch of crates it currently uses that certain stacks devs are not happy with. Once the SDK is approved for stacks blockchain use, hopefully we can remove this duplicated code living on 3989.
